tools/ds-identify

```diff
@@ -1409,33 +1376,31 @@ dscheck_VMware() {
 
     # Check to see if running in a container and the VMware
     # datasource is configured via environment variables.
-    if vmware_has_envvar_vmx_guestinfo; then
-        if vmware_has_envvar_vmx_guestinfo_metadata || \
-            vmware_has_envvar_vmx_guestinfo_userdata || \
-            vmware_has_envvar_vmx_guestinfo_vendordata; then
+    if [ -n "${VMX_GUESTINFO}" ]; then
+        if [ -n "${VMX_GUESTINFO_METADATA}" ] || \
+           [ -n "${VMX_GUESTINFO_USERDATA}" ] || \
+           [ -n "${VMX_GUESTINFO_VENDORDATA}" ]; then
             return "${DS_FOUND}"
         fi
     fi
 
     # Do not proceed unless the detected platform is VMware.
-    if [ ! "${DI_VIRT}" = "vmware" ]; then
-        return "${DS_NOT_FOUND}"
-    fi
+    [ "${DI_VIRT}" = "vmware" ] || return ${DS_NOT_FOUND}
 
     # Do not proceed if the vmware-rpctool command is not present.
-    if ! vmware_has_rpctool; then
-        return "${DS_NOT_FOUND}"
-    fi
+    command -v vmware-rpctool >/dev/null 2>&1 || return ${DS_NOT_FOUND}
 
     # Activate the VMware datasource only if any of the fields used
     # by the datasource are present in the guestinfo table.
-    if { vmware_rpctool_guestinfo_metadata || \
-         vmware_rpctool_guestinfo_userdata || \
-         vmware_rpctool_guestinfo_vendordata; } >/dev/null 2>&1; then
-        return "${DS_FOUND}"
-    fi
+    local tmp=''
+    for tmp in "metadata" "userdata" "vendordata"; do
+      tmp="$(vmware-rpctool "info-get guestinfo.$tmp")"
+      if [ $? -eq 0 ]; then
+          echo "$tmp" | grep -q "[[:alnum:]]" && return ${DS_FOUND}
```

Comment:
Please change this to `"${DS_FOUND}"` to be consistent with how other values are used. Quoting variables in BASH is preferred unless otherwise required.

Reply:
DS_FOUND and DS_NOT_FOUND are integers here, no problem.

Reply:
It does not matter if they are integers. In fact, POSIX integer comparison requires quoting, ex. `[ "${x}" -gt "1" ]`. Unless needing explicit expansion, shell variables should almost always be quoted based on my experience.

Reply:
Done